dcc-release-job/dcc-release-job-id/src/main/java/org/icgc/dcc/release/job/id/task/AddSurrogateMutationIdTask.java

```diff
@@ -1,41 +1,156 @@
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
 
 public class AddSurrogateMutationIdTask extends AddSurrogateIdTask {
+  private static final String ASSEMBLY_VERSION = "GRCh37";
+  private static final String MUTATION_ID_PREFIX = "MU";
+  public static final String mutationDumpPath = "/pg_dump/mutation/mutation.txt";
+  private DataFrame mutationIDs;
+  private SQLContext sqlContext;
 
-  public AddSurrogateMutationIdTask(@NonNull IdClientFactory idClientFactory) {
+  public AddSurrogateMutationIdTask(@NonNull IdClientFactory idClientFactory, DataFrame df, SQLContext sqlContext) {
     super(FileType.SSM_P_MASKED, FileType.SSM_P_MASKED_SURROGATE_KEY, idClientFactory);
+    this.mutationIDs = df;
+    this.sqlContext = sqlContext;
   }
 
   @Override
   protected JavaRDD<ObjectNode> process(JavaRDD<ObjectNode> input) {
-    return input
-        .map(new AddSurrogateMutationId(idClientFactory));
+
+    IdClientFactory localIdClientFactory = idClientFactory;
+
+    DataFrame raw_df =
+      sqlContext.createDataFrame(
+        input.map(row -> {
+          String chromosome = row.get(SUBMISSION_OBSERVATION_CHROMOSOME).textValue();
+          String chromosomeStart = textValue(row, SUBMISSION_OBSERVATION_CHROMOSOME_START);
+          String chromosomeEnd = textValue(row, SUBMISSION_OBSERVATION_CHROMOSOME_END);
+          String mutation = row.get(NORMALIZER_MUTATION).textValue();
+          String mutationType = row.get(SUBMISSION_OBSERVATION_MUTATION_TYPE).textValue();
+          String assemblyVersion = ASSEMBLY_VERSION;
+          return new MutationID(chromosome, chromosomeStart, chromosomeEnd, mutation, mutationType, assemblyVersion, "");
+        }),
+        MutationID.class
+      );
+
+
+    String[] fields = {"chromosome", "chromosomeStart", "chromosomeEnd", "mutation", "mutationType", "assemblyVersion"};
+
+    return
+        raw_df.join(
+            mutationIDs.withColumnRenamed("uniqueId", "db.uniqueId"),
+            WrapAsScala$.MODULE$.asScalaBuffer(Arrays.asList(fields)),
+            "left_outer"
+        ).rdd().toJavaRDD()
+        .map(row -> {
+
+          String chromosome = row.<String>getAs("chromosome");
+          String chromosomeStart = row.<String>getAs("chromosomeStart");
+          String chromosomeEnd = row.<String>getAs("chromosomeEnd");
+          String mutation = row.<String>getAs("mutation");
+          String mutationType = row.<String>getAs("mutationType");
+          String assemblyVersion = row.<String>getAs("assemblyVersion");
+
+          if(row.<String>getAs("db.uniqueId") == null || row.<String>getAs("db.uniqueId").isEmpty()){
+            return
+              new MutationID(
+                  chromosome,
+                  chromosomeStart,
+                  chromosomeEnd,
+                  mutation,
+                  mutationType,
+                  assemblyVersion,
+                  localIdClientFactory.create().createMutationId(
```

Comment:
leaving a reminder here to change this so a client isn't instantiated for each missing mutationId.